packages/app/src/routers/SavedviewsRouter/components/ColorOptionComponents.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  DisplayStyle3dState,
+  EmphasizeElements,
+  IModelApp,
+} from "@bentley/imodeljs-frontend";
+import { ColorPickerButton } from "@bentley/ui-components";
+import { Button, ExpandableBlock, ToggleSwitch } from "@itwin/itwinui-react";
+import React, { useState } from "react";
+
+import { ComponentsColorPicker } from "./ComponentInputs";
+import "./ComponentInputs.scss";
+import { generateJSONInStyles } from "./GenerateJSONStyles";
+let SelectedElementColor = ColorDef.fromString("skyblue");
+
+export interface EnvironmentDisplayToggleProps {
+  label: string;
+  environmentType: string;
+  path: any[];
+  pathLength: number;
+}
+function getEnvironmentDisplay(environmentType: string): boolean {
+  const newBackground = IModelApp.viewManager.selectedView?.displayStyle.toJSON();
+  return (
+    newBackground?.jsonProperties?.styles?.environment?.[environmentType]
+      ?.display ?? false
+  );
+}
+export function EnvironmentDisplayToggle({
+  label,
+  environmentType,
+  path,
+  pathLength,
+}: EnvironmentDisplayToggleProps) {
+  const [toggleValue, setToggleValue] = useState<boolean>(
+    getEnvironmentDisplay(environmentType)
+  );
+
+  return (
+    <ToggleSwitch
+      className="text-setting"
+      label={label}
+      checked={toggleValue}
+      onChange={() => {
+        const newBackground = IModelApp.viewManager.selectedView?.displayStyle
+          .clone()
+          .toJSON();
+
+        if (pathLength !== path.length) {
+          path.pop();
+        }
+
+        path.push(!toggleValue);
+
+        if (newBackground !== undefined) {
+          generateJSONInStyles(newBackground, path);
+        }
+
+        const vp = IModelApp.viewManager.selectedView;
+
+        if (vp !== undefined && newBackground !== undefined) {
+          const state = new DisplayStyle3dState(newBackground, vp.iModel);
+          state
+            .load()
+            .then(() => {
+              vp.view.setDisplayStyle(state);
+            })
+            .catch((ex) => {
+              console.log(`Error found: ${ex}`);
+            });
+        }
+
+        setToggleValue(!toggleValue);
+      }}
+    />
+  );
+}
+
+export interface SelectedElementColorPickerProps {
+  label: string;
+}
+export function SelectedElementColorPicker({
+  label,
+}: SelectedElementColorPickerProps) {
+  return (
+    <div className="label-with-color-picker">
+      <h3 className="text-setting">{label}</h3>
+      <ColorPickerButton
+        initialColor={ColorDef.fromString("skyblue")}
+        onColorPick={(color: ColorDef) => {
+          SelectedElementColor = color;
+        }}
+      />
+    </div>
+  );
+}
+
+export interface SelectedElementColorButtonProps {
+  label: string;
+  action: string;
+}
+export function SelectedElementColorButton({
+  label,
+  action,
+}: SelectedElementColorButtonProps) {
+  return (
+    <div className="button-container text-setting">
+      <Button
+        className="button-label"
```

Comment:
Use `size='small'` instead of more css

Reply:
replaced the css class with size setting